Source/WebCore/dom/ContainerNode.cpp

```diff
@@ -86,21 +86,25 @@ unsigned ScriptDisallowedScope::s_count = 0;
 ScriptDisallowedScope::EventAllowedScope* ScriptDisallowedScope::EventAllowedScope::s_currentScope = nullptr;
 #endif
 
-ALWAYS_INLINE auto ContainerNode::removeAllChildrenWithScriptAssertion(ChildChange::Source source, NodeVector& children, DeferChildrenChanged deferChildrenChanged) -> DidRemoveElements
+ALWAYS_INLINE auto ContainerNode::removeAllChildrenWithScriptAssertion(ChildChange::Source source, NodeVector& children, DeferChildrenChanged deferChildrenChanged) -> RemoveAllChildrenResult
 {
     ASSERT(children.isEmpty());
     collectChildNodes(*this, children);
 
+    auto childrenDeletionCanBeDelayed = ChildrenDeletionCanBeDelayed::Yes;
+
     if (UNLIKELY(isDocumentFragmentForInnerOuterHTML())) {
         ScriptDisallowedScope::InMainThread scriptDisallowedScope;
         RELEASE_ASSERT(!connectedSubframeCount() && !hasRareData() && !wrapper());
         bool hadElementChild = false;
+        int treeSize = 0;
         while (RefPtr child = m_firstChild.get()) {
+            treeSize++;
```

Comment:
This is clearly not right. Each removed node can have more descendants.
Traversing the node tree is expensive here so we don't wanna do that either.
We should probably return 0 or `std::nullopt` for `treeSize` to signify
to the caller that we don't know how many nodes are there.

Reply:
Note that we know for sure this code wouldn't end up resulting in node removal so it's totally okay to bail out.